cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/sql/SqlStructuredTableRegistry.java

```diff
@@ -190,6 +192,6 @@ public void drop(StructuredTableId tableId) {
           throw exception;
         }
       };
-    return new SqlTransactionRunner(specAdmin, dataSource);
+    return new SqlTransactionRunner(specAdmin, dataSource, new NoOpMetricsCollectionService(), false);
```

Comment:
Need to pass the real collection service here, instead of the no-op one, right?

Reply:
Same here, I think when we use the registry class to register for storage creation, the MetricsCollectionService might not get started

Reply:
Same here, add comment